src/categorical_algebra/FinCats.jl

```diff
@@ -303,18 +367,39 @@ function FinCatPresentation(pres::Presentation{ThSchema})
   FinCatPresentation{ThSchema,Ob,Hom}(pres)
 end
 
+function FinCatPresentation(pres::Presentation{ThPtSchema})
+  S = pres.syntax
+  Ob = Union{S.Ob, S.AttrType}
+  Hom = Union{S.Hom, S.Attr, S.AttrType}
+  FinCatPresentation{ThPtSchema,Ob,Hom}(pres)
+end
+"""
+The opposite of a FinCatPresentation whose syntax is not just FreeCategory
+is effectively interpreted as if it were, since the opposite may not 
+strictly speaking exist in another syntax.
+"""
+function hom_generators(C::OppositeCat{Ob,Hom,FinCatSize,Ct}) where {Ob,Hom,Ct<:FinCatPresentation}
```

Comment:
I'm confused by why this is needed. If you use the categories API in the intended way, always interpreting Julia objects that represent objects and morphisms *relative to a category* (`Cat` instance), then taking the opposite category should never require munging the data of the original category.

Reply:
I think the problem is that hom generators in a FinCatPresentation know their domain and codomain. Previously `hom_generators(op(C))` would return `hom_generators(C)`, which makes `dom` and `codom` be wrong.